types/react-navigation/index.d.ts

```diff
@@ -651,3 +652,22 @@ export interface NavigationScreenProps<Params> {
 /**
  * END CUSTOM CONVENIENCE INTERFACES
  */
+
+
+/*
```

Comment:
Good point, that section naming is a bit dated now that other things have been put in there, so I agree that `CUSTOM CONVENIENCE INTERFACES` is a bit inaccurate. You can leave this as-is for now, and maybe we can reorganize everything in a separate PR later. Thoughts on labeling sections either just as `Header`, `StackNavigator`, etc., versus by corresponding `react-navigation`-repo file names (`src/views/HeaderBackButton`)?

For the former, that's pretty simple, but if we want to point to the `src/...` file in a comment, it will look kinda messy. On the other hand, for the latter, there might be cases where a definition doesn't quite fit under a single file name.

Just thinking out loud, not important for this PR, but open to any opinions now. Thanks!